src/main/kotlin/service/UserService.kt

```diff
@@ -60,4 +64,24 @@ class UserService(
         users.forEach { builder.addUsers(it.toGrpcUser()) }
         return builder.build()
     }
+
+    /**
+     * Service implementation of [SnowballRService.softDeleteUser].
+     */
+    override suspend fun softDeleteUser(request: Base.Id): Base.Nothing {
+        val currentUser = repo.getUserById(dummyUserId!!)
+        val userToDelete = repo.getUserById(request.id)
+
+        // Checks, if the user tries to delete another user that is admin (not possible even if current user is admin)
+        if (userToDelete.role == UserRole.USER_ROLE_ADMIN && currentUser.id.toString() != request.id) {
+            throw FailedPreconditionException.Admin(userToDelete.id.toString())
+        }
+        // Checks, if the user tries to delete another user without being an admin
+        if (currentUser.role != UserRole.USER_ROLE_ADMIN && currentUser.id.toString() != request.id) {
+            throw UnauthorizedException.All.User(dummyUserId!!)
+        }
+
+        repo.softDeleteUser(parseUUID(request.id, "user"))
```

Comment:
Use the object that you already have.
```suggestion
        repo.softDeleteUser(userToDelete.id)
```